### PR TITLE
Ticket 6 post comment for article

### PR DESCRIPTION
**Pull request by MuseOfCode**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -5,15 +5,18 @@ const data = require("../db/data/test-data");
...
-    // console.log("DB SEEDED???");
+    console.log("Seeding complete");
   });
 });
 
 afterAll(() => {
-  return db.end();
+  return db.end().then(() => {
+    console.log("Database connection closed");
+  });
```
Comment (ncjaycarroll) [inline __tests__/app.test.js:19]: Is there a need for these console.logs ?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -190,60 +200,119 @@ describe("GET /api/articles/:article_id/comments", () => {
...
       .then(({ body }) => {
-        expect(body.msg).toBe("Endpoint not found");
+        expect(body.comment).toMatchObject({
+          comment_id: expect.any(Number),
+          article_id: 3,
+          author: "lurker",
+          body: "Code is fun... until it isn’t. Then it's just debugging.",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
```
Comment (ncjaycarroll) [inline __tests__/app.test.js:231]: The votes and comment_id are things you can know for sure in your assertions

File: __tests__/app.test.js [hunk 3]
```diff
@@ -190,60 +200,119 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+
+  test("Status: 201, Ignores extra fields in the request body", () => {
     return request(app)
-      .get("/api/articles/invalid_id/comments")
-      .expect(400)
+      .post("/api/articles/3/comments")
+      .send({
+        username: "lurker",
+        body: "Code is fun... until it isn’t. Then it's just debugging.",
+        extra_field: "should be ignored",
```
Comment (ncjaycarroll) [inline __tests__/app.test.js:241]: Good consideration of edge cases

File: __tests__/app.test.js [hunk 4]
```diff
@@ -190,60 +200,119 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+          author: "lurker",
+          body: "Code is fun... until it isn’t. Then it's just debugging.",
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+        expect(body.comment).not.toHaveProperty("extra_field");
       });
   });
-  test("Status: 404, Responds with 404 - Not found if article_id is invalid", () => {
+  test("Status: 201, Responds with a posted comment on articles that have 0 comments", () => {
```
Comment (ncjaycarroll) [inline __tests__/app.test.js:256]: What functionality does this test really test for that is different from previous tests?

File: models/comments.model.js [hunk 5]
```diff
@@ -7,11 +8,37 @@ exports.fetchCommentsByArticleId = (article_id) => {
...
+            );
+          }
+        })
+        .then(({ rows }) => {
+          return rows[0];
+        });
+    })
+    .catch((err) => {
+      return Promise.reject(err);
+    });
```
Comment (ncjaycarroll) [inline models/comments.model.js:43]: Don't have catch blocks in the model, have them in the controllers attached to the invocation of the model function, that way the comment can be passed back to app.js with next()
